standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -3905,4 +3905,29 @@ void createOrDropTriggerToPoolMapping(String resourcePlanName, String triggerNam
    * @return String representation of the version number of Metastore server (eg: 3.1.0-SNAPSHOT)
    */
   String getServerVersion() throws TException;
+
+  /**
+   * Returns details about a scheduled query by name.
+   * 
+   * @throws NoSuchObjectException if an object by the given name dosen't exists.
+   */
+  ScheduledQuery getScheduledQuery(ScheduledQueryKey scheduleKey) throws TException;
+
+  /**
+   * Carries out maintenance of scheduled queries (insert/update/drop).
+   */
+  void scheduledQueryMaintenance(ScheduledQueryMaintenanceRequest request) throws MetaException, TException;
+
+  /**
+   * Checks whenever a query is available for execution.
+   *
+   * @return optionally a scheduled query to be processed.
+   */
+  ScheduledQueryPollResponse scheduledQueryPoll(ScheduledQueryPollRequest request) throws MetaException, TException;
```

Comment:
I wonder whether we should be able to pull multiple queries at once? Maybe a possible extension... we can create a follow-up.

Reply:
you mean multiple scheduled query executors in the same hs2?